Cargo.toml

```diff
@@ -11,49 +11,249 @@ readme = "README.md"
-rand_core = { version = "0.6.4", default-features = false, features = ["getrandom"] }
-rsa = { version = "0.9.2", default-features = false, features = ["sha2"] }
-rustls = { version = "0.23.12", default-features = false }
-sec1 = { version = "0.7.3", default-features = false, features = ["pkcs8", "pem"] }
-sha2 = { version = "0.10.7", default-features = false }
-signature = { version = "2.1.0", default-features = false }
-webpki = { package = "rustls-webpki", version = "0.102.0", default-features = false }
-x25519-dalek = { version = "2", default-features = false }
+# Cryptographic dependencies
+aead = { version = "0.6.0-rc.2", default-features = false, optional = true }
+aes = { version = "0.9.0-rc.1", default-features = false, optional = true }
+aes-gcm = { version = "0.11.0-rc.1", default-features = false, optional = true }
+ccm = { version = "0.6.0-pre", default-features = false, optional = true, git = "https://github.com/RustCrypto/AEADs/" }
+chacha20 = { version = "0.10.0-rc.2", default-features = false, optional = true }
+chacha20poly1305 = { version = "0.11.0-rc.1", default-features = false, optional = true }
+cipher = "0.5.0-rc.1"
+crypto-common = { version = "0.2.0-rc.4", default-features = false }
+der = { version = "0.8.0-rc.9", default-features = false, optional = true }
+digest = { version = "0.11.0-rc.3", default-features = false }
+ecdsa = { version = "0.17.0-rc.7", default-features = false, optional = true }
+ed25519-dalek = { version = "3.0.0-pre.1", default-features = false, optional = true }
+ed448-goldilocks = { version = "0.14.0-pre.4", default-features = false, optional = true }
+elliptic-curve = { version = "0.14.0-rc.15", default-features = false, optional = true }
+hmac = { version = "0.13.0-rc.2", default-features = false }
+p256 = { version = "0.14.0-pre.11", default-features = false, optional = true }
+p384 = { version = "0.14.0-pre.11", default-features = false, optional = true }
+p521 = { version = "0.14.0-pre.11", default-features = false, optional = true }
+pkcs1 = { version = "0.8.0-rc.4", default-features = false, optional = true }
+pkcs8 = { version = "0.11.0-rc.7", default-features = false, optional = true }
+rsa = { version = "0.10.0-rc.9", default-features = false, optional = true }
+sec1 = { version = "0.8.0-rc.10", default-features = false, optional = true }
+sha2 = { version = "0.11.0-rc.2", default-features = false }
+signature = { version = "3.0.0-rc.4", default-features = false, optional = true }
+typenum = { version = "1.18.0", features = ["no_std", "const-generics"] }
+x25519-dalek = { version = "3.0.0-pre.1", default-features = false, optional = true }
+x448 = { version = "=0.14.0-pre.0", default-features = false, optional = true }
+
+# External groups
+pki-types = { package = "rustls-pki-types", version = "1.12.0", default-features = false }
+rand_core = { version = "0.9.3", default-features = false, features = [
+    "os_rng",
```

Comment:
os_rng should be optional. in no-std environments this will not often work where custom has to be provided.